rpc/types.go

```diff
@@ -35,10 +35,14 @@ type API struct {
 	Public    bool        // indication if the methods must be considered safe for public use
 }
 
-// Error wraps RPC errors, which contain an error code in addition to the message.
-type Error interface {
-	Error() string  // returns the message
-	ErrorCode() int // returns the code
+// ErrorWithCode defines an error code in addition to the message.
+type ErrorWithCode interface {
+	Code() int // returns the code
+}
+
+// ErrorWithData defines a data item in addition to the message.
+type ErrorWithData interface {
+	Data() interface{} // returns the error data
```

Comment:
I don't like this change at all. The previous definitions were fine, why do we need to change them now?

The `Code` and `Data` method names say nothing. That's why I chose `ErrorCode`. If you see a method definition for `ErrorCode`, you'll know what it means. `Code` is just not specific enough.